hydra-explorer/src/Hydra/Explorer/ExplorerState.hs

```diff
@@ -0,0 +1,197 @@
+module Hydra.Explorer.ExplorerState where
+
+import Hydra.Prelude
+
+-- XXX: Depending on hydra-node will be problematic to support versions
+import Hydra.HeadId (HeadId (..))
+
+import Hydra.Cardano.Api (TxIn, TxOut)
+import Hydra.Cardano.Api.Prelude (CtxUTxO)
+import Hydra.Chain.Direct.Tx (
+  AbortObservation (..),
+  CloseObservation (..),
+  CollectComObservation (..),
+  CommitObservation (..),
+  ContestObservation (..),
+  FanoutObservation (..),
+  HeadObservation (..),
+  InitObservation (..),
+ )
+import Hydra.ContestationPeriod (ContestationPeriod)
+import Hydra.OnChainId (OnChainId)
+import Hydra.Party (Party)
+
+data PartyCommit = PartyCommit
+  { txIn :: TxIn
+  , txOut :: TxOut CtxUTxO
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Arbitrary PartyCommit where
+  arbitrary = genericArbitrary
+
+data HeadMember = HeadMember
+  { party :: Party
+  , onChainId :: OnChainId
+  , commits :: [PartyCommit]
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Arbitrary HeadMember where
+  arbitrary = genericArbitrary
+
+data HeadStatus
```

Comment:
A bit annoying that we have similar but not exacly the same type in the `ServerOutput.HeadStatus`